kyo-prelude/jvm/src/test/scala/kyo2/kernel/SafepointTest.scala

```diff
@@ -244,7 +244,7 @@ class SafepointTest extends Test:
                     yield 42
                 }.eval
 
-                assert(count == 6)
```

Comment:
This is a related fix. The `Effect.defer` method doesn't need to call `Safepoint.handle` since it's executed right after the evaluation decides that the computation can continue. These `Effect.defer` frames are also synthetic, not generated from user code, so I think it makes sense not to have them in the generated traces.